predicators/envs/burger.py

```diff
@@ -1376,17 +1409,44 @@ def create_tasks_for_type_two(
                 patty2 = Object("patty2", self._patty_type)
                 state_dict[patty2] = {"row": r, "col": c, "z": 0}
                 hidden_state[patty2] = {"is_cooked": 0.0, "is_held": 0.0}
+
+                if i < CFG.burger_num_test_start_holding:
+                    # replace the cheese with a bottom bun
+                    cheese1 = [o for o in state_dict if o.name == "cheese1"][0]
+                    state_dict.pop(cheese1)
+                    hidden_state.pop(cheese1)
+                    r, c = shuffled_spots[4]  # where the cheese was
+                    bottom_bun2 = Object("bottom_bun2", self._bottom_bun_type)
+                    state_dict[bottom_bun2] = {"row": r, "col": c, "z": 0}
+                    hidden_state[bottom_bun2] = {"is_held": 0.0}
+                    # add another patty
+                    r, c = shuffled_spots[8]  # next empty cell
+                    patty3 = Object("patty3", self._patty_type)
+                    state_dict[patty3] = {"row": r, "col": c, "z": 0}
+                    hidden_state[patty3] = {"is_cooked": 0.0, "is_held": 0.0}
+                    # start out holding patty3.
+                    state_dict[self._robot]["fingers"] = 1.0
+                    state_dict[patty3]["row"] = state_dict[self._robot]["row"]
+                    state_dict[patty3]["col"] = state_dict[self._robot]["col"]
+                    hidden_state[patty3] = {"is_cooked": 0.0, "is_held": 1.0}
+
                 test_goal = {
                     GroundAtom(self._IsCooked, [patty]),
                     GroundAtom(self._On, [patty, bottom_bun]),
                     GroundAtom(self._On, [patty2, patty]),
                     GroundAtom(self._On, [top_bun, patty2])
                 }
+                if i < CFG.burger_num_test_start_holding:
+                    test_goal.add(GroundAtom(self._IsCooked, [patty3]))
+                    test_goal.add(GroundAtom(self._On, [patty3, bottom_bun2]))
                 alt_test_goal = {
                     GroundAtom(self._GoalHack2, [bottom_bun, patty]),
                     GroundAtom(self._GoalHack6, [patty, patty2]),
                     GroundAtom(self._On, [top_bun, patty2])
                 }
+                if i < CFG.burger_num_test_start_holding:
+                    alt_test_goal.add(
+                        GroundAtom(self._GoalHack2, [bottom_bun2, patty3]))
```

Comment:
just to confirm my understanding here - is the goal when `i < CFG.burger_num_test_start_holding` that the agent should cook the patty it's holding and also put it on a bottom bun (in addition to the usual goal)? Why do we need to explicitly tell it to cook the thing it's holding - can we just not change the goal at all? Just curious

Reply:
The goal is to have a flag that lets you control what % of test tasks have you start out holding a patty. The easy way to implement this was to have the first `N=CFG.burger_num_test_start_holding` test tasks have this property. 

My initial thought was:
- We need the patty to be cooked before the agent places it on the bottom bun because our "place" operator that is applicable here has a goalhack add effect that is "patty is cooked + patty is above bottom bun". 

But actually -- we don't have any goal related to that extra patty we start out holding, I don't think it should be a problem taht we don't actually satisfy the add effects of the operator upon execution, since we aren't doing execution monitoring at the moment. 

Can continue discussion on slack.

Reply:
Ok -- if we remove explicit goals related to the patty we start out holding, we still find a successful plan that starts out with placing that patty (and we cook it before placing it, as expected).